src/main/java/seedu/address/commons/util/FileUtil.java

```diff
@@ -123,6 +124,28 @@ public static boolean hasExtension(Path filePath, String[] allowedExtensions) {
                 .reduce(false, (x, y) -> x || y);
     }
 
+    /**
+     * Checks if the content of the file is prefixed with any magic number from a given set.
+     *
+     * @throws IOException If the file cannot be read.
+     */
+    public static boolean hasMagicNumber(Path filePath, Set<Byte[]> allowedMagicNumber) throws IOException {
+        byte[] bytes = Files.readAllBytes(filePath);
+        Byte current;
+        for (Byte[] magicNumber : allowedMagicNumber) {
+            for (int i = 0; i < magicNumber.length; i++) {
+                current = magicNumber[i];
+                if (current != null && bytes[i] != current) {
+                    break;
+                }
+                if (i == magicNumber.length - 1 && (current == null || current == bytes[i])) {
+                    return true;
+                }
```

Comment:
If i'm not wrong this section can be simplified to:
```java
if (current != null && bytes[i] != current) {
    break;
} else if (i == magicNumber.length - 1) {
    return true;
}
```

Reply:
A null byte represents a wildcard for file signature and may be necessary if more file signature is added in the future